src/Microsoft.ApplicationInsights.AspNetCore/Extensions/ApplicationInsightsExtensions.cs

```diff
@@ -201,12 +201,28 @@ public static IServiceCollection AddApplicationInsightsTelemetry(this IServiceCo
                 // that requires IOptions infrastructure to run and initialize
                 services.AddSingleton<IStartupFilter, ApplicationInsightsStartupFilter>();
 
-                services.AddSingleton<JavaScriptSnippet>();
-                services.AddSingleton<ApplicationInsightsLoggerEvents>();
+                services.AddSingleton<JavaScriptSnippet>();                
 
                 services.AddOptions();
                 services.AddSingleton<IOptions<TelemetryConfiguration>, TelemetryConfigurationOptions>();
                 services.AddSingleton<IConfigureOptions<TelemetryConfiguration>, TelemetryConfigurationOptionsSetup>();
+
+                // NetStandard2.0 has a package reference to Microsoft.Extensions.Logging.ApplicationInsights, and
+                // enables ApplicationInsightsLoggerProvider by default.
+                // This also adds LoggerMarker to the DI Container, which is used by LoggerFactory.AddApplicationInsights() method
+                // to detect ApplicationInsightsLoggerProvider as already added and back-off.
+                // Without this, there will be double logging if LoggerFactory.AddApplicationInsights() was already used.
+#if NETSTANDARD2_0
+                services.AddLogging(loggingBuilder =>
+                {
+                     loggingBuilder.AddApplicationInsights();
+
+                     // By default, all logs Warning or above is captured.
+                     // AddFilter is additive
+                     loggingBuilder.AddFilter<Microsoft.Extensions.Logging.ApplicationInsights.ApplicationInsightsLoggerProvider>("Default", LogLevel.Warning);
```

Comment:
I wonder if we are messing with the ILogger infra here. This can already be done from the config which is read as part of ILogger setup. Hiding this statement this deep in SDK might be non-intuitive.

Reply:
I'd be interested in @davidfowl's opinion here. Ideally, custom middleware authors would wire up all of the services requisite for their expected behavior. If someone uses `AddApplicationInsights`, for instance, we'd presume all trace, exception - everything, by default - goes to Application Insights, and that any service wire-ups required for that goal to be "easy" or even "invisible" to the developer.

Reply:
Since users updatng to this new SDK version has not explicitly made a decision to send ILogger logs to AI, i'd like to just sent Warning or above to AI. Without this, we could end up sending too much logs to AI. (negative surprises when billing comes). Since the runtime itself logs using ILogger, we have experienced user complaints about too much AI logs. (https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/603)

I'd open to all suggestions here. We want to balance between 'easy/auto capture of ILogger logs' and 'too much noise'.

Reply:
By default, only "Warning" or above LogLevel logs are sent to ApplicationInsightsProvider for all categories. But this can be easily configured by user either in code or using appsettings.json.  
I added more description here: https://github.com/Microsoft/ApplicationInsights-aspnetcore/issues/854